core/src/main/kotlin/com/justai/jaicf/activator/Activator.kt

```diff
@@ -54,15 +55,18 @@ interface Activator {
      *
      * @param botContext a current user's [BotContext]
      * @param request a current [BotRequest]
+     * @param activationSelector a [ActivationSelector] implementation to select activation
      * @return [Activation] that contains an optional state of scenario and [com.justai.jaicf.context.ActivatorContext] or null if activator cannot handle a request at all.
      *
      * @see BotContext
      * @see BotRequest
      * @see com.justai.jaicf.context.ActivatorContext
+     * @see ActivationSelector
      */
     fun activate(
         botContext: BotContext,
-        request: BotRequest
+        request: BotRequest,
+        activationSelector: ActivationSelector
```

Comment:
Are you sure that selector should be passed to `Activator` for each `activate` invocation? Could it be any case where selector could be changed at runtime?

Reply:
It doesn't. But I don't like adding `ActivationSelector` as property to `Activator` interface. Not every `Activator` needs `activationSelector` as property. In fact, `Activator` implementation can operate without `ActivationSelector`. It's `Activator`'s implementation decision to use `ActivationSelector` or not. 

Adding `ActivationSelector` as property means that we should add it to `ActivatorFactory`, therefore making implementing optional logic even more strict.

So I decided to add activationSelector as argument. UPD: With default value.